### PR TITLE
Adjust import paths 

### PR DESCRIPTION
**Pull request by Disiok**
# Description
- reduce root level imports (breaking change)
- make `llama_index.indices` consistent, actually expose all indices (only addition)
- move `llama_index.indices.postprocessor` to `llama_index.postprocessor` (keep old for backward compatibility)
- move `llama_index.indices.query.schema.*` to `llama_index.schema` (keep old for backward compatibility)
- create `llama_index.core` and move `BaseQueryEngine`, `BaseRetriever` (keep old for backward compatibility)
- move `llama_index.indices.service_context` to `llama_index.service_context` (keep old for backward compatibility)

### CODE REVIEW

File: llama_index/__init__.py [hunk 1]
```diff
@@ -151,28 +88,11 @@
     # indices
     "SummaryIndex",
     "VectorStoreIndex",
-    "SimpleKeywordTableIndex",
```
Comment (logan-markewich) [inline llama_index/__init__.py:154]: I think moving the keyword, knowledge graph, and document summary index imports will cause some issues.

File: llama_index/__init__.py [hunk 2]
```diff
@@ -151,28 +88,11 @@
...
-    "PandasIndex",
-    "EmptyIndex",
-    "DocumentSummaryIndex",
-    "KnowledgeGraphIndex",
     # indices - legacy names
-    "GPTKeywordTableIndex",
-    "GPTKnowledgeGraphIndex",
-    "GPTSimpleKeywordTableIndex",
-    "GPTRAKEKeywordTableIndex",
-    "GPTListIndex",
```
Comment (logan-markewich) [inline llama_index/__init__.py:168]: Curious -- why remove some GPT prefixes and keep others?

File: llama_index/__init__.py (hunk 1 above)
Comment (Disiok) [reply llama_index/__init__.py:154]: my thought was to keep the most common ones covering >90 what people are using. 

good pushback, I'll add those back.

File: llama_index/__init__.py (hunk 2 above)
Comment (logan-markewich) [reply llama_index/__init__.py:168]: I would keep also GPTListIndex, others are probably fine to remove the legacy import

Comment (Disiok) [reply llama_index/__init__.py:168]: yep

**Pull request merged**